pkg/proxy/proxy.go

```diff
@@ -145,6 +145,110 @@ func (p *SprayProxy) HandleProxy(c *gin.Context) {
 	c.String(http.StatusOK, "proxied")
 }
 
+func (p *SprayProxy) HandleProxyEndpoint(c *gin.Context) {
+	// currently not distinguishing between requests we can parse and those we cannot parse
+	metrics.IncInboundCount()
+	errors := []error{}
+	zapCommonFields := []zapcore.Field{
+		zap.String("method", c.Request.Method),
+		zap.String("path", c.Request.URL.Path),
+		zap.String("query", c.Request.URL.RawQuery),
+		zap.Bool("insecure-tls", p.insecureTLS),
+	}
+	// Read in body from incoming request
+	buf := &bytes.Buffer{}
+	_, err := buf.ReadFrom(c.Request.Body)
+	defer c.Request.Body.Close()
+	if err != nil {
+		c.String(http.StatusRequestEntityTooLarge, "too large: %v", err)
+		p.logger.Error("request body too large", zapCommonFields...)
+		return
+	}
+	body := buf.Bytes()
+
+	client := &http.Client{}
+	if p.insecureTLS {
+		client.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		}
+	}
+
+	for _, backend := range p.backends() {
+		backendURL, err := url.Parse(backend)
+		if err != nil {
+			p.logger.Error("failed to parse backend "+err.Error(), zapCommonFields...)
+			continue
+		}
+		copy := c.Copy()
+		newURL := copy.Request.URL
+		newURL.Host = backendURL.Host
+		newURL.Scheme = backendURL.Scheme
```

Comment:
A potential bug here - the copied URL will maintain `/proxy` in its path. This needs to be dropped. Otherwise the proxy will forward to `https://<backend-host>/proxy/`. Tests should verify that `/proxy` is dropped from the path.

Reply:
Fixed